app/javascript/packs/document-capture.jsx

```diff
@@ -1,10 +1,15 @@
 import React from 'react';
 import { render } from 'react-dom';
 import DocumentCapture from '../app/document-capture/components/document-capture';
+import I18nContext from '../app/document-capture/context/i18n';
+
+const { I18n: i18n } = window.LoginGov;
```

Comment:
> Is changing case like this common in react? In my mind, we'd either always use lower or upper case for it?

In this case, it's something where I'd have conventionally expected the key to have been named `i18n` instead of `I18n`, and in a way this was meant to "correct" it specifically to its usage here.

Reference: https://github.com/airbnb/javascript#naming--camelCase

Ideally it'd be something where the original key of `LoginGov` would be renamed, instead of assigning it this way here. Alternatively, we could choose to stick with `I18n` and avoid the changing case here. There's probably not much of a downside to it. In other cases, it could become a concern if the non-camelCase would pervade through multiple files of the application, where it might be ideal to try to adjust at the top-level entrypoint, like done here.

See also https://github.com/18F/identity-idp/pull/3911#discussion_r453900328, where there could be some justification for using PascalCase as a convention for enums. That said, the Airbnb standards make no mention of this.

Reply:
> Actually on second thought.... IMO wouldn't this be clearer since we're only doing a single assign?

Possibly, yeah. As mentioned above, it was done more this way as an explicit intent to rename, though in hindsight there might be a better way to go about this, or to consider the locally scoped variable as more of a pointer than a renamed, destructured key of the object.

>Or is there a react shim we can do for this (not the hook) so we can just do:

I'm not sure I follow this one. What are you envisioning would be the contents of `some_custom_react_module` ?

Reply:
my imagined `some_custom_react_module` would be something that pulls from the `LoginGov` global variable (or equivalent) so we don't to mix global variables and these mostly modularized components? And that way if we choose to import translations another way for React, we could encapsulate that behind this module interface

Reply:
And that module could also transform the casing into a nice way once? So that all the React modules that use it don't have to do this conversion?

Reply:
> my imagined `some_custom_react_module` would be something that pulls from the `LoginGov` global variable (or equivalent) so we don't to mix global variables and these mostly modularized components? And that way if we choose to import translations another way for React, we could encapsulate that behind this module interface

If I'm understanding correctly, this is what I was hoping to go for with trying to isolate references to these global variables from this file, treating this `packs` entry point as the integration point between the application, where it's reasonable to have awareness of things like specific DOM elements or window globals. Contrasted with the rest of the React app source, where we'd seek to avoid these references in favor of relatively generic, reusable components. It would be expected that the entry point does a minimal initialization of all relevant values via props or context. Once the value is assigned to the context provider, it can just be consumed as the given object, with no need to reference the global anymore.

Is that on the same line of thinking that you had in mind?

I could see it being further generalized in that localization isn't an exclusive concern of document capture, but instead something we'd likely reuse between different domains.

>And that module could also transform the casing into a nice way once? So that all the React modules that use it don't have to do this conversion?

With what's implemented here, there shouldn't need to be any references to `window.LoginGov` within the `document-capture` directory, or at least I would hope to avoid this.

Reply:
Ok! Maybe that was the bit that confused me -- I wasn't looking for access to `window.LoginGov` inside that directory, I was looking at it in this file --- that makes sense! Zero react experience here so I probably need to do some more homework on the directory layout -- thanks for explaining!

Reply:
> Zero react experience here so I probably need to do some more homework on the directory layout -- thanks for explaining!

I'm still navigating the Webpacker folder structure conventions, so certainly welcome any feedback here as well. In trying to decipher the role of the `packs` entrypoint file as somehow distinct from the rest of the application source, it felt to me an appropriate place to do this sort of necessary bootstrapping / integration logic.